File: libs/python/pyrepository/pyrepository/interfaces/ingestors/dtos.py
from dataclasses import dataclass


@dataclass
class JobMetadataParams:
    _id: str
    name: str
    active: bool
    frequency: str
    context: str
    source: str
    service: str

@dataclass
class JobParams:
    jobHandler: str
    url: str

@dataclass
class Config:
    jobMetadataParams: JobMetadataParams
    jobParams: JobParams

@dataclass
class MessageMetadata:
    processing_id: str
    processing_timestamp: str
    source: str
    service: str

@dataclass
class MessageParameters:
    id: str
    data: dict
    metadata: MessageMetadata


@dataclass
class MessageMetadataOriginsOutput:
    gateway: str
    controller: str

@dataclass
class MessageMetadataOutput:
    id: str
    data: dict
    processing_id: str
    processing_timestamp: str
    source: MessageMetadataOriginsOutput

@dataclass
class MetadataOutput:
    input: MessageMetadataOutput
    service: MessageMetadataOriginsOutput
    processing_id: str
    processing_timestamp: str
    target_endpoint: str
    job_frequency: str

@dataclass
class MessageStatus:
    code: int
    detail: str

File: apps/lake-orchestration/lake-gateway/internal/usecase/dtos.go
package usecase

type Metadata struct {
	ProcessingId        string `json:"processing_id"`
	ProcessingTimestamp string `json:"processing_timestamp"`
	Source              string `json:"source"`
	Service             string `json:"service"`
}

type Status struct {
	Code   int    `json:"code"`
	Detail string `json:"detail"`
}

type InputInputDTO struct {
	Data map[string]interface{} `json:"data"`
}

type InputOutputDTO struct {
	ID       string                 `json:"id"`
	Data     map[string]interface{} `json:"data"`
	Metadata Metadata               `json:"metadata"`
	Status   Status                 `json:"status"`
}

type InputStatusInputDTO struct {
	ID     string `json:"id"`
	Status Status `json:"status"`
}

type ListInputOutputDTO struct {
	Inputs []InputOutputDTO `json:"inputs"`
}

type StagingJobInputDTO struct {
	InputId      string                 `json:"input_id"`
	Input        map[string]interface{} `json:"input"`
	Source       string                 `json:"source"`
	Service      string                 `json:"service"`
	ProcessingId string                 `json:"processing_id"`
}

type StagingJobOutputDTO struct {
	ID           string                 `json:"id"`
	InputId      string                 `json:"input_id"`
	Input        map[string]interface{} `json:"input"`
	Source       string                 `json:"source"`
	Service      string                 `json:"service"`
	ProcessingId string                 `json:"processing_id"`
}
